enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -220,8 +222,10 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
           new DefaultRepositorySystemSession(session);
 
       // For netty-handler referencing its dependencies with ${os.detected.classifier}
-      fullDependencyResolutionSession.setSystemProperties(
```

Comment:
This was removing system properties such as "java.version", resulting in profiles not resolvable (#817 )